Assets/Scripts/Combat/Abilities/DragAndDrop.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class DragAndDrop : MonoBehaviour, IBeginDragHandler, IDragHandler, IEndDragHandler
+{
+    
+    private Vector3 originalPos;
+    private bool isInside = false;
+    private RectTransform rectTransform;
+    private CanvasGroup canvasGroup;
+
+    [SerializeField] private Canvas canvas;
+
+    private void Start()
+    {
+        rectTransform = GetComponent<RectTransform>();
+        canvasGroup = GetComponent<CanvasGroup>();
+        originalPos = rectTransform.localPosition;
+    }
+
+    public void ResetPosition()
+    {
+        rectTransform.localPosition = originalPos;
+        isInside = false;
+    }
+
+    public void OnBeginDrag(PointerEventData eventData)
+    {
+        canvasGroup.blocksRaycasts = false;
+    }
+
+    public void OnDrag(PointerEventData eventData)
+    {
+        // TODO: possibly snap back to original position ?
+        if(!isInside) // for sweets/rots
+            rectTransform.anchoredPosition += eventData.delta;
+    }
+
+    public void OnEndDrag(PointerEventData eventData)
+    {
+        canvasGroup.blocksRaycasts = true;
+    }
+
+    public void IsInside(bool b)
+    {
+        isInside = b;
+    }
+
+    public bool GetInside()
+    {
+        return isInside;
+    }
+}
```

Comment:
newline at EOF  pls

Reply:
I'm not sure what you mean by this

Reply:
Nevermind, I understand. What does the newline change ?